### PR TITLE
PTFM-38976 Update clone/move CLI to accept targetFileRelocation flag.

### PR DESCRIPTION
**Pull request by austinlocke**
https://jira.internal.dnanexus.com/browse/PTFM-38976
PTFM-38976 Update clone/move CLI to accept targetFileRelocation flag.

### CODE REVIEW

**Review by arpradhan — APPROVED** [review]

**Review by haroldna — APPROVED** [review]
lgtm 🌮

File: src/python/dxpy/cli/cp.py [hunk 1]
```diff
@@ -151,7 +151,8 @@ def cp(args):
                                     {"objects": src_objects,
                                      "folders": src_folders,
                                      "project": dest_proj,
-                                     "destination": dest_path})['exists']
+                                     "destination": dest_path,
+                                     "targetFileRelocation": str(args.target_file_relocation).lower()})['exists']
```
Comment (kpjensen) [inline src/python/dxpy/cli/cp.py:155]: Is the field always safe to provide as input regardless of the file being a symlink?

File: src/python/dxpy/scripts/dx.py [hunk 2]
```diff
@@ -4803,6 +4805,7 @@ def positive_number(value):
                                            nargs='+')
 cp_sources_action.completer = DXPathCompleter()
 parser_cp.add_argument('destination', help=fill('Folder into which to copy the sources or new pathname (if only one source is provided).  Must be in a different project/container than all source paths.', width_adjustment=-15))
+parser_cp.add_argument('--target-file-relocation', help='Flag to allow symlink target file relocation in external storage while cloning a symlink.', default=False)
```
Comment (kpjensen) [inline src/python/dxpy/scripts/dx.py:4808]:
```suggestion
parser_cp.add_argument('--target-file-relocation', help='Allow symlink target file relocation in external storage while cloning a symlink.', default=False)
```

File: src/python/dxpy/scripts/dx.py [hunk 3]
```diff
@@ -4818,6 +4821,7 @@ def positive_number(value):
                                            nargs='+')
 mv_sources_action.completer = DXPathCompleter()
 parser_mv.add_argument('destination', help=fill('Folder into which to move the sources or new pathname (if only one source is provided).  Must be in the same project/container as all source paths.', width_adjustment=-15))
+parser_mv.add_argument('--target-file-relocation', help='Flag to allow symlink target file relocation in external storage while moving a symlink.', default=False)
```
Comment (kpjensen) [inline src/python/dxpy/scripts/dx.py:4824]:
```suggestion
parser_mv.add_argument('--target-file-relocation', help='Allow symlink target file relocation in external storage while moving a symlink.', default=False)
```

**Review by kpjensen — CHANGES_REQUESTED** [review]

File: src/python/dxpy/cli/cp.py (hunk 1 above)
Comment (austinlocke) [reply src/python/dxpy/cli/cp.py:155]: Yes there is logic on the apiserver side to check for it.

File: src/python/dxpy/scripts/dx.py [hunk 4]
```diff
@@ -4803,6 +4805,7 @@ def positive_number(value):
                                            nargs='+')
 cp_sources_action.completer = DXPathCompleter()
 parser_cp.add_argument('destination', help=fill('Folder into which to copy the sources or new pathname (if only one source is provided).  Must be in a different project/container than all source paths.', width_adjustment=-15))
+parser_cp.add_argument('--target-file-relocation', help='Allow symlink target file relocation in external storage while cloning a symlink.', default=False)
```
Comment (kpjensen) [inline src/python/dxpy/scripts/dx.py:4808]:
```suggestion
parser_cp.add_argument('--target-file-relocation', help='Allow symlink target file relocation in external storage while cloning a symlink.', default=False, action='store_true')```

File: src/python/dxpy/scripts/dx.py [hunk 5]
```diff
@@ -4818,6 +4821,7 @@ def positive_number(value):
                                            nargs='+')
 mv_sources_action.completer = DXPathCompleter()
 parser_mv.add_argument('destination', help=fill('Folder into which to move the sources or new pathname (if only one source is provided).  Must be in the same project/container as all source paths.', width_adjustment=-15))
+parser_mv.add_argument('--target-file-relocation', help='Allow symlink target file relocation in external storage while moving a symlink.', default=False)
```
Comment (kpjensen) [inline src/python/dxpy/scripts/dx.py:4824]:
```suggestion
parser_mv.add_argument('--target-file-relocation', help='Allow symlink target file relocation in external storage while moving a symlink.', default=False, action='store_true')
```

**Review by kpjensen — CHANGES_REQUESTED** [review]

**Review by kpjensen — APPROVED** [review]

**Pull request merged**